lib/Test2/Mock.pm

```diff
@@ -203,7 +203,7 @@ package $class;
         if (\$self->SUPER::can(\$meth)) {
             return \$self->SUPER::can(\$meth);
         }
-        elsif (exists \$self->{\$meth}) {
+        elsif (ref \$self eq 'HASH' && exists \$self->{\$meth}) {
```

Comment:
I could change this into `reftype` if loading Scalar::Util in here is acceptable

Reply:
Loading Scalar::Util is totally acceptable, please do.